crates/matrix-sdk-common/src/runtime.rs

```diff
@@ -0,0 +1,112 @@
+//! that can be used in place of tokio's `Handle` and `Runtime` types.
+
+#[cfg(not(target_family = "wasm"))]
+mod sys {
+    /// A handle to a runtime for executing async tasks and futures.
+    pub type Handle = tokio::runtime::Handle;
+    pub type Runtime = tokio::runtime::Runtime;
+}
+
+#[cfg(target_family = "wasm")]
+mod sys {
+    use std::future::Future;
+
+    use super::super::executor::{spawn, JoinHandle};
+
+    /// A handle to a runtime for executing async tasks and futures.
+    pub type Handle = WasmRuntimeHandle;
+    pub type Runtime = WasmRuntimeHandle;
+
+    #[derive(Debug)]
+    /// A dummy guard that does nothing when dropped.
+    /// This is used for the Wasm implementation to match
+    /// tokio::runtime::EnterGuard.
+    pub struct WasmRuntimeGuard;
+
+    impl Drop for WasmRuntimeGuard {
+        fn drop(&mut self) {
+            // No-op, as there's no special context to exit in Wasm
+        }
+    }
+
+    #[derive(Default, Debug)]
+    /// A runtime handle implementation for WebAssembly targets.
+    ///
+    /// This implements a minimal subset of the tokio::runtime::Handle API
+    /// that is needed for the matrix-rust-sdk to function on Wasm.
+    pub struct WasmRuntimeHandle;
+
+    impl WasmRuntimeHandle {
+        /// Spawns a future in the wasm32 bindgen runtime.
+        #[track_caller]
+        pub fn spawn<F>(&self, future: F) -> JoinHandle<F::Output>
+        where
+            F: Future + 'static,
+            F::Output: 'static,
+        {
+            spawn(future)
+        }
+
+        /// Runs the provided function on an executor dedicated to blocking
+        /// operations.
+        #[track_caller]
+        pub fn spawn_blocking<F, R>(&self, func: F) -> JoinHandle<R>
+        where
+            F: FnOnce() -> R + 'static,
+            R: 'static,
+        {
+            spawn(async move { func() })
+        }
+
+        /// Runs a future to completion on the current thread.
+        pub fn block_on<F, T>(&self, future: F) -> T
+        where
+            F: Future<Output = T>,
+        {
+            futures_executor::block_on(future)
+        }
+
+        /// Enters the runtime context.
+        ///
+        /// For WebAssembly, this is a no-op that returns a dummy guard.
+        pub fn enter(&self) -> WasmRuntimeGuard {
+            WasmRuntimeGuard
+        }
+    }
+}
+
+/// Get a runtime handle appropriate for the current target platform.
+///
+/// This function returns a unified `Handle` type that works across both
+/// Wasm and non-Wasm platforms, allowing code to be written that is
+/// agnostic to the platform-specific runtime implementation.
+///
+/// Returns:
+/// - A `tokio::runtime::Handle` on non-Wasm platforms
+/// - A `WasmRuntimeHandle` on Wasm platforms
+pub fn get_runtime_handle() -> Handle {
+    #[cfg(target_family = "wasm")]
+    return Handle::default();
+
+    #[cfg(not(target_family = "wasm"))]
+    async_compat::get_runtime_handle()
```

Comment:
I believe you can remove the `return` here:

```suggestion
    #[cfg(target_family = "wasm")]
    {
        Handle::default()
    }

    #[cfg(not(target_family = "wasm"))]
    {
        async_compat::get_runtime_handle()
    }
```